assets/src/components/form/inputs/base/option-checked-state.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External imports
+ */
+import { isArray, indexOf } from 'lodash';
+import PropTypes from 'prop-types';
+
+/**
+ * generates the initial checked state
+ * for a set of checkbox or radio button
+ * options based on the set defaultValue.
+ * This can be overridden by supplying
+ * an alternate value via the fourth param,
+ * which is used to reset all values.
+ *
+ * @function
+ * @param {string} inputId
+ * @param {Array} options
+ * @param {mixed} values
+ * @param {mixed} reset
+ * @return {Object} state
+ */
+export const OptionCheckedState = ( inputId, options, values, reset ) => {
+	const checked = {};
+	options = isArray( options ) ? options : [];
+	values = isArray( values ) ? values : [ values ];
+	options.forEach( function( option, index ) {
+		const key = inputId + '-' + index;
+		const isValue = indexOf( values, option.value ) !== -1;
```

Comment:
no because it's not determining whether the set of options *has* a selected value or not (or more than one for multi value inputs), it is determining whether the current option *is* the selected value (or one of the selected values).

for example: if the question was "favourite colour" and the options were "red", "orange", "yellow", "green", "blue", "purple", and the currently selected value was "orange", then as we loop through these options, all of the options except "orange" would have a value of `-1` for `isValue` because they would not be in the array of selected `values`.

so based on that explanation, maybe a better name would be `isSelected` ?

I'm open to suggestions if you can come up with something even better.

PS: nitpicking is good when it results in better code ;)

Reply:
ya I think `isSelectedValue` or `isSelected` would work better here.  `isValue` makes it sound like  you're determining whether it is a value.  So being a bit more precise in the naming will help here.